### DISCUSSION TITLE
srsran 4G ZMQ errors...tx time is 2.000 ms in the past

### DISCUSSION THREAD

**Discussion by ItsGarand** [srsran/srsRAN_Project#997] [General Help]
Greetings and Happy Holidays to all that observe..

I have srsran 4G set up and running on Rocky Linux 9 using the ZMQ virtual radios and a gnuradio-compnaion graph. The first test seemingly ran without any issues using the suggested / out of the box configuration that was outlined in the docs.  The EPC, eNb, and UE all ran, the ping test worked, and I got gobs of IQ data in the output sinks.  So I guess all good.

When I changed the number of resource blocks to 25 and the sample rate to 7.68e6 (enb and ue) I get the to following error spilling out of the eNb:
 
[zmq] Error: tx time is 2.000 ms in the past...

Here is the output of the eNb:

CH0 rx_port=tcp://localhost:2100
CH0 tx_port=tcp://*:2101
CH0 fail_on_disconnect=true

==== eNodeB started ===
Type <t> to view trace
Error: couldn't update sample rate. 5.76 is not divisible by 7.68
Current sample rate is 1.92 MHz with a base rate of 7.68 MHz (x4 decimation)
Error: couldn't update sample rate. 5.76 is not divisible by 7.68
Current sample rate is 1.92 MHz with a base rate of 7.68 MHz (x4 decimation)
Setting frequency: DL=2680.0 Mhz, UL=2560.0 MHz for cc_idx=0 nof_prb=25
[zmq] Error: tx time is 2.000 ms in the past (38400 < 53760)
[zmq] Error: tx time is 2.000 ms in the past (84480 < 99840)
[zmq] Error: tx time is 2.000 ms in the past (130560 < 145920)
[zmq] Error: tx time is 2.000 ms in the past (176640 < 192000)
[zmq] Error: tx time is 4.000 ms in the past (222720 < 253440)
[zmq] Error: tx time is 2.000 ms in the past (268800 < 284160)
[zmq] Error: tx time is 2.000 ms in the past (314880 < 330240)

This is the output of the ue:

Active RF plugins: libsrsran_rf_zmq.so
Inactive RF plugins: 
Couldn't open , trying /root/.config/srsran/ue.conf
Reading configuration file /root/.config/srsran/ue.conf...

Built in Release mode using commit ec29b0c1f on branch master.

Opening 1 channels in RF device=zmq with args=tx_port=tcp://*:2001,rx_port=tcp://localhost:2000,id=ue,base_srate=7.68e6
Supported RF device list: zmq file
CHx base_srate=7.68e6
CHx id=ue
Current sample rate is 1.92 MHz with a base rate of 7.68 MHz (x4 decimation)
CH0 rx_port=tcp://localhost:2000
CH0 tx_port=tcp://*:2001
Waiting PHY to initialize ... done!
Attaching UE...
Current sample rate is 1.92 MHz with a base rate of 7.68 MHz (x4 decimation)
Current sample rate is 1.92 MHz with a base rate of 7.68 MHz (x4 decimation)
...................................................................................................................^CStopping

So, the question is why am I seeing the tx time errors and where is the sample rate of 5.76 coming from?

NOTE: I don't see this issue when I set the number of resource blocks to 15 and the sample rate to 3.84e6

There was a topic similar to this in the mail archives, but I did not find the solution...

**Comment by pgawlowicz** [comment]
Please use the sampling rate of 5.76e6, or add the following to your enb and ue configs:

```
[expert]
lte_sample_rates = true
```

Please note that this discussion list is dedicated to the srsRAN_Project.
